pkg/apis/kubermatic/v1/settings.go

```diff
@@ -59,17 +59,17 @@ type SettingSpec struct {
 	// +kubebuilder:default=false
 
 	// EnableWebTerminal enables the Web Terminal feature for the user clusters.
-	EnableWebTerminal bool `json:"enableWebTerminal,omitempty"`
+	EnableWebTerminal *bool `json:"enableWebTerminal,omitempty"`
 
 	// +kubebuilder:default=true
 
 	// EnableShareCluster enables the Share Cluster feature for the user clusters.
-	EnableShareCluster bool `json:"enableShareCluster,omitempty"`
+	EnableShareCluster *bool `json:"enableShareCluster,omitempty"`
 
 	EnableOIDCKubeconfig bool `json:"enableOIDCKubeconfig"` //nolint:tagliatelle
 
 	// DisableAdminKubeconfig disables the admin kubeconfig functionality on the dashboard.
-	DisableAdminKubeconfig bool `json:"disableAdminKubeconfig,omitempty"`
+	DisableAdminKubeconfig *bool `json:"disableAdminKubeconfig,omitempty"`
```

Comment:
We only need pointer for `EnableShareCluster`, not for the rest IMO.

Reply:
I wanted to keep the type consistent for all bool fields with `omitempty` but you are right. To resolve the problem we need it only for `EnableShareCluster`.

Reply:
My thought process was that by introducing a consistent type we reduce the risk of making same mistake in the future.

Reply:
I'd like to disagree, `EnableShareCluster` is an exception and should be treated as one. For the other fields since the functionality works with primitive types and without the need for pointers, I'd prefer to keep it that way.

Personal opinion, I would have just removed the defaulting to true for `EnableShareCluster` since I don't think it's really required. A lot of our customers have this disabled. But yes, that's a subjective opinion and we shouldn't do this in code freeze probably.